.github/workflows/main.yml

```diff
@@ -0,0 +1,240 @@
+            build-options: ""
+
+    name: "Build - ${{ matrix.artifact-name }}"
+    runs-on: ubuntu-latest
+    container: ${{ matrix.container }}
+    steps:
+      - run: sudo apt-get update && sudo apt-get install -y gfortran 
+        name: install gfortran
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+      - run: sudo apt-get update && sudo apt-get install ant -y && sudo rm -rf /var/lib/apt/lists/* && sudo rm -f /bin/ant && sudo ln -s /usr/share/ant/bin/ant /bin/ant
+        name: Install Ant
+      - run: rm -rf /usr/local/arm-linux-gnueabihf && curl -SL https://github.com/wpilibsuite/opensdk/releases/download/v2023-9/armhf-raspi-bullseye-2023-x86_64-linux-gnu-Toolchain-10.2.0.tgz | sh -c 'mkdir -p /usr/local && cd /usr/local && tar xzf - --strip-components=2'
+        name: Replace arm32 compiler
+        if: matrix.artifact-name == 'Arm32'
+      - uses: lukka/run-vcpkg@v11.5
+        with:
+          runVcpkgInstall: true
+          vcpkgGitCommitId: '29b2ea2d4b6197e66ef346e62ccbba35b55b7de5'
+      - run: vcpkg install
+        name: install packages
+      - run: ./gradlew publish ${{ matrix.build-options }}
+        name: Build with Gradle
+      - uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.artifact-name }}
+          path: gradleDir/outputs/
+
+  build-windows:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - artifact-name: Win64Release
+            architecture: x64
+            build-options: "-PskipDebug"
+          - artifact-name: Win32Release
+            architecture: x86
+            build-options: "-PskipDebug"
+          - artifact-name: WinArm64Release
+            architecture: x64
+            build-options: "-PskipDebug -Pplatform=windows-arm64"
+          - artifact-name: Win64
+            architecture: x64
+            build-options: "-PskipRelease -PskipSources"
+          - artifact-name: Win32
+            architecture: x86
+            build-options: "-PskipRelease -PskipSources"
+          - artifact-name: WinArm64
+            architecture: x64
+            build-options: "-PskipRelease -PskipSources -Pplatform=windows-arm64"
+
+    name: "Build - ${{ matrix.artifact-name }}"
+    runs-on: windows-2019
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+      - uses: actions/setup-java@v4
+        with:
+          java-version: 11
+          distribution: 'zulu'
+          architecture: ${{ matrix.architecture }}
+      - name: Set Java Heap Size
+        run: sed -i 's/-Xmx2g/-Xmx1g/g' gradle.properties
+        if: matrix.architecture == 'x86'
+      - uses: lukka/run-vcpkg@v11.5
+        with:
+          runVcpkgInstall: true
+          vcpkgGitCommitId: '29b2ea2d4b6197e66ef346e62ccbba35b55b7de5'
+      - run: vcpkg install
+        name: install packages
+      - run: |
+          cd vcpkg_installed
+          ls
+        name: show
+      - run: |
```

Comment:
Use the `ilammy/msvc-dev-cmd` action instead of calling the vcvars64 script. Also, since vcpkg is being invoked by the workflow instead of Gradle, `ilammy/msvc-dev-cmd` needs the arch to be specified.